Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs

```diff
@@ -644,34 +642,7 @@ void UpdateLeftBarButtonItem(ParentingViewController containerController, Page p
 			if ((firstPage != pageBeingRemoved && currentChild != firstPage && NavigationPage.GetHasBackButton(currentChild)) || _parentMasterDetailPage == null)
```

Comment:
This expression seems complicated, and it probably makes sense to not complicate it even further. So I refactored below code into its own method and let phone master detail renderer call it freely.